homa_impl.h

```diff
@@ -2962,7 +2962,7 @@ static inline __be32 ip6_as_be32(const struct in6_addr x)
 	if (is_mapped_ipv4(x)) {
 		return x.in6_u.u6_addr32[3];
 	} else {
-		return x.in6_u.u6_addr32[3];
+		return x.in6_u.u6_addr32[1];
```

Comment:
I switched this to pick [3] because, at least in my test cluster, all the interesting information is in the last word of the IPv6 address; if I use [1], all the addresses show up in timetraces as 0. Is it different at Google (is there some convention that my cluster isn't following)?

Reply:
For a global addres like 2600:1702:1d00:f836::4e7 the index [1] will be nonzero and [2] will be zero.

For a link local address like fe80::1193:dd16:8475:f03e the index [1] will be zero. 

I guess we should extract a different set of bits depending on if it's a link-local address or not.

Reply:
Is [3] a bad choice? I was assuming that the low-order bits are the ones most likely to differentiate hosts, but everything I know about IPv7 addresses I've learned since last Friday (which is not much).

Reply:
On many systems the last 32 bits are all zeros.